src/prefect/environments/execution/fargate/fargate_task.py

```diff
@@ -0,0 +1,279 @@
+import os
+from typing import Any, List
+
+import cloudpickle
+
+import prefect
+from prefect import config
+from prefect.environments.execution import Environment
+from prefect.environments.storage import Docker
+
+# from boto3 import client as boto3_client
+# from botocore.exceptions import ClientError
+
+
+class FargateTaskEnvironment(Environment):
+    """
+    FargateTaskEnvironment is an environment which deploys your flow (stored in a Docker image)
+    as a Fargate tasl. This environment requires AWS credentials and extra boto3 kwargs which
+    are used in the creation and running of the Fargate task.
+
+    When providing a custom container definition spec the first container in the spec must be the
+    container that the flow runner will be executed on.
+
+    These environment variables are required for cloud do not need to be included because
+    they are instead automatically added and populated during execution:
+
+    - `PREFECT__CLOUD__GRAPHQL`
+    - `PREFECT__CLOUD__AUTH_TOKEN`
+    - `PREFECT__CONTEXT__FLOW_RUN_ID`
+    - `PREFECT__CONTEXT__NAMESPACE`
+    - `PREFECT__CONTEXT__IMAGE`
+    - `PREFECT__CONTEXT__FLOW_FILE_PATH`
+    - `PREFECT__CLOUD__USE_LOCAL_SECRETS`
+    - `PREFECT__ENGINE__FLOW_RUNNER__DEFAULT_CLASS`
+    - `PREFECT__ENGINE__TASK_RUNNER__DEFAULT_CLASS`
+    - `PREFECT__LOGGING__LOG_TO_CLOUD`
+
+    Additionally, the following command will be applied to the first container:
+
+    `$ /bin/sh -c 'python -c "from prefect.environments import FargateTaskEnvironment; FargateTaskEnvironment().run_flow()"'`
+
+    All `kwargs` are accepted that one would normally pass to boto3 for `register_task_definition`
+    and `run_task`. For information on the kwargs supported visit the following links:
+
+    https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ecs.html#ECS.Client.register_task_definition
+
+    https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ecs.html#ECS.Client.run_task
+
+    The secrets and kwargs that are provided at initialization time of this environment
+    are not serialized and will only ever exist on this object.
+
+    Args:
+        - aws_access_key_id (str, optional): AWS access key id for connecting the boto3
+            client. Defaults to the value set in the environment variable
+            `AWS_ACCESS_KEY_ID`.
+        - aws_secret_access_key (str, optional): AWS secret access key for connecting
```

Comment:
Should we also expose a method of pulling these from Prefect Secrets? jw

Reply:
Like an optional flag to say whether or not to grab these from a Secret?

Reply:
Would make sense but the initialization will happen locally when building the environment

Reply:
Hmm more like:
```
- aws_credentials_secret (str, optional)
```
like on our AWS tasks which refers to a Secret name that is only pulled at runtime.  We don't have to do that for this PR but something to maybe think about.

Reply:
Yeah let's put that in a future enhancement, after this is merged I'll open an issue for it. I like the idea